### PR TITLE
Assignments 2.4

### PR DESCRIPTION
**Pull request by elohanlon**

### CODE REVIEW

File: app/assets/javascripts/digital_objects_app/widgets/digital_object_captions_editor.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+      // TODO: Don't use Hyacinth.DigitalObjectsApp.currentUser as an indication of whether we're in the js app or not. There's a better way. Temp solution here.
+      url: this.assignment && (!Hyacinth.DigitalObjectsApp.currentUser) ? '/assignments/' + this.assignment['id'] + '/changeset/proposed' : '/digital_objects/' + this.digitalObject.getPid() + '/captions',
+    },
+    options: {
+      previewOnly: this.mode == 'view'
+    }
+  };
+
+	this.synchronizerWidget = new OHSynchronizer(widgetOptions);
+  OHSynchronizer.playerControls.bindNavControls(); //bind modal forward/back/etc. nav controls. TODO: Move this to widget js instead of Hyacinth js
```
Comment (barmintor) [inline app/assets/javascripts/digital_objects_app/widgets/digital_object_captions_editor.js]: is there a ticket for this TODO?

File: app/assets/javascripts/digital_objects_app/widgets/digital_object_captions_editor.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  OHSynchronizer.errorHandler = function(e) {
+    Hyacinth.addAlert(e, 'danger');
+  }
+
+  this.$containerElement.find('.save-captions-button').on('click', $.proxy(this.saveCaptions, this));
+
+  // TODO: Move code below into synchronizer widget?
```
Comment (barmintor) [inline app/assets/javascripts/digital_objects_app/widgets/digital_object_captions_editor.js]: Is there a ticket for this TODO?

File: app/controllers/assignments/changesets_controller.rb [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+          create_or_update_synchronize_changeset(@assignment, digital_object, captions_params[:captions_text])
+          @assignment.save
+        else
+          raise 'Not implemented yet'
+        end
+      end
+    when 'describe'
+      if description_params[:digital_object_data_json]
+        # in order to run cleanup on submitted dynamic field data, we'll run it through the normal digital object data update method
+        # TODO: In next version of Hyacinth, separate out the cleanup code so that we don't need to rely on a digital object instance
```
Comment (barmintor) [inline app/controllers/assignments/changesets_controller.rb]: ticket for this TODO?

File: app/controllers/assignments_controller.rb [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+  def new
+  end
+
+  # POST /assignments
+  def create
+    @assignment.status = 'unassigned' if @assignment.assignee.blank?
+
+    # original and proposed fields should both be set upon assignment creation
+    case @assignment.task
+    # TODO: Add 'describe', 'annotate', and 'sequence' types to case statement
```
Comment (barmintor) [inline app/controllers/assignments_controller.rb]: annotate and describe are here; ticket the sequence assignment and update comment?

File: app/assets/javascripts/digital_objects_app/widgets/digital_object_captions_editor.js (hunk 1 above)
Comment (elohanlon) [reply app/assets/javascripts/digital_objects_app/widgets/digital_object_captions_editor.js]: There is now! LDPD-109

File: app/assets/javascripts/digital_objects_app/widgets/digital_object_captions_editor.js (hunk 2 above)
Comment (elohanlon) [reply app/assets/javascripts/digital_objects_app/widgets/digital_object_captions_editor.js]: It is now! LDPD-110

File: app/controllers/assignments/changesets_controller.rb (hunk 3 above)
Comment (elohanlon) [reply app/controllers/assignments/changesets_controller.rb]: HYACINTH-267

File: app/controllers/assignments_controller.rb (hunk 4 above)
Comment (elohanlon) [reply app/controllers/assignments_controller.rb]: Updated!

**Pull request merged**